bioagents/qca/qca.py

```diff
@@ -0,0 +1,292 @@
+# QCA stands for qualitative causal agent whose task is to
+# identify causal paths within
+
+import os
+import logging
+#from indra.statements import ActiveForm
+#from indra.bel.processor import BelProcessor
+import json
+import ndex.client as nc
+import requests
+import io
+
+logger = logging.getLogger('QCA')
+
+_resource_dir = os.path.dirname(os.path.realpath(__file__)) + '/../resources/'
+
+class PathNotFoundException(Exception):
+    def __init__(self, *args, **kwargs):
+            Exception.__init__(self, *args, **kwargs)
+
+class QCA:
+    def __init__(self):
+        logger.debug('Starting QCA')
+        self.host = "http://www.ndexbio.org"
```

Comment:
You could consider passing these (host names, credentials, etc) as arguments on startup that can be integrated into the central system startup script. This way these potentially sensitive details won't need to be hard coded and published.